presto-main/src/test/java/com/facebook/presto/operator/TestHashJoinOperator.java

```diff
@@ -320,31 +394,57 @@ public void testInnerJoinWithSpill(boolean probeHashEnabled, List<WhenSpill> whe
         }
     }
 
+    private static void checkErrors(final TaskStateMachine taskStateMachine)
+            throws Throwable
+    {
+        if (taskStateMachine.getFailureCauses().size() > 0) {
+            throw taskStateMachine.getFailureCauses().peek();
+        }
+    }
+
     private enum WhenSpill
     {
         DURING_BUILD, AFTER_BUILD, DURING_USAGE, NEVER
     }
 
+    private enum WhenSpillFails
+    {
+        NEVER, SPILL_HBO, SPILL_LJO, UNSPILL_HBO, UNSPILL_LJO
+    }
+
     @DataProvider
     public Object[][] testInnerJoinWithSpillDataProvider()
     {
-        List<Object[]> result = new ArrayList<>();
+        return joinWithSpillParameters(true).stream().map(List::toArray).toArray(Object[][]::new);
```

Comment:
Usually, one would put `.map` and `.toArray` on separate lines (retaining `.stream()` where it is)